### PR TITLE
Make sonic_sfp Python2 and Python3 compatible

### PR DESCRIPTION
**Pull request by msosyak**
Make `_read_eeprom_specific_bytes` Python3 and Python2 compatible

### CODE REVIEW

File: sonic_platform_base/sonic_sfp/sfputilbase.py [hunk 1]
```diff
@@ -338,8 +338,15 @@ def _read_eeprom_specific_bytes(self, sysfsfile_eeprom, offset, num_bytes):
             return None
 
         try:
-            for n in range(0, num_bytes):
-                eeprom_raw[n] = hex(ord(raw[n]))[2:].zfill(2)
+            # in case raw is bytes (python3 is used) raw[n] will return int,
+            # and in case raw is str(python2 is used) raw[n] will return str,
+            # so for python3 the are no need to call ord to convert str to int.
```
Comment (jleveque) [inline sonic_platform_base/sonic_sfp/sfputilbase.py:343]: Thanks for this fix. Can you also please add a `#TODO: Remove this check once we no longer support Python 2` comment here for posterity?

**Review by jleveque — CHANGES_REQUESTED** [review]

**Review by jleveque — CHANGES_REQUESTED** [review]
I noticed there is also a call to `self.eep_dict.iteritems()` on line 746. Can you also please change this to `self.eep_dict.items()` to make it compatible with both Python 2 and 3?

**Comment by msosyak** [comment]
> I noticed there is also a call to `self.eep_dict.iteritems()` on line 746. Can you also please change this to `self.eep_dict.items()` to make it compatible with both Python 2 and 3?

Done

File: sonic_platform_base/sonic_sfp/sfputilbase.py (hunk 1 above)
Comment (msosyak) [reply sonic_platform_base/sonic_sfp/sfputilbase.py:343]: Done

**Review by jleveque — APPROVED** [review]

**Pull request merged**